journalists/app/src/main/java/com/csatimes/dojmajournalists/Activity/AddVideoActivity.java

```diff
@@ -0,0 +1,182 @@
+        super.onCreate(savedInstanceState);
+        mAuth = FirebaseAuth.getInstance();
+        setContentView(R.layout.activity_add_video);
+
+        creator = findViewById(R.id.creator);
+
+        String[] departments = new String[]{"DoJMA", "DoPY"};
+        ArrayAdapter<String> creator_adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, departments);
+        creator.setAdapter(creator_adapter);
+        creator.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                switch (i) {
+                    case 0:
+                        creator_final = "DoJMA";
+                        break;
+                    case 1:
+                        creator_final = "DoPY";
+                        break;
+
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+
+        type = findViewById(R.id.source);
+
+        String[] types = new String[]{"youtube", "facebook"};
+        ArrayAdapter<String> type_adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_dropdown_item, types);
+        type.setAdapter(type_adapter);
+        type.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                switch (i) {
+                    case 0:
+                        chosen_type = "youtube";
+                        break;
+                    case 1:
+                        chosen_type = "facebook";
+                        break;
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+
+        videoName = findViewById(R.id.video_title);
+        description = findViewById(R.id.video_desc);
+        videoUrl = findViewById(R.id.url);
+
+        video_upload_date = findViewById(R.id.date);
+        addBtn = findViewById(R.id.add);
+        progressBar = findViewById(R.id.progress_bar);
+        progressBar.setVisibility(View.GONE);
+        video_upload_date.setOnClickListener(v -> {
+            video_upload_date.setError(null);
+            final DatePickerDialog dpd;
+            final Calendar c = Calendar.getInstance();
+            final int currentYear = c.get(Calendar.YEAR);
+            final int currentMonth = c.get(Calendar.MONTH);
+            final int currentDate = c.get(Calendar.DAY_OF_MONTH);
+            dpd = new DatePickerDialog(AddVideoActivity.this,
+                    (view, year, month, dayOfMonth) ->
+                            video_upload_date.setText(String.format(Locale.ENGLISH, "%02d-%02d-%04d",
+                                    dayOfMonth, month+1, year)),
```

Comment:
See [Simple Date Format](https://developer.android.com/reference/java/text/SimpleDateFormat) for easier way to do this.

Reply:
Umm.. ya that makes more sense, but I just copied code from other activites. This implementation is also effective. Not required to change as far as I think.